src/shared/components/ProfilePage/Stats/DistributionGraph/index.jsx

```diff
@@ -0,0 +1,288 @@
+import ChartTooltip from '../ChartTooltip';
+import './index.scss';
+
+const getRanges = distribution => (
+  toPairs(distribution)
+    .map(([name, number]) => ({
+      name,
+      number,
+      start: parseInt(name.match(/\d+/)[0], 10),
+      end: parseInt(name.match(/To(\d+)/)[1], 10),
+    }))
+    .sort((a, b) => a.start - b.start)
+);
+
+function removeLeadingZeroes(ranges) {
+  while (ranges[0].number === 0) {
+    ranges.shift();
+  }
+}
+
+function removeTrailingZeroes(ranges) {
+  while (ranges[ranges.length - 1].number === 0) {
+    ranges.pop();
+  }
+}
+
+function ratingToColor(colors, rating) {
+  const filteredColors = colors.filter(color => rating >= color.start && rating <= color.end);
+  return (filteredColors[0] && filteredColors[0].color) || 'black';
+}
+
+export default class DistributionGraph extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {};
+    this.graphRef = React.createRef();
+  }
+
+  componentDidMount() {
+    const $scope = this;
+    $scope.desktop = window.innerWidth >= 900;
+    this.draw();
+    this.resizeHandle = () => {
+      if (window.innerWidth < 900 && $scope.desktop) {
+        $scope.desktop = false;
+        this.draw();
+      } else if (window.innerWidth >= 900 && !$scope.desktop) {
+        $scope.desktop = true;
+        this.draw();
+      }
+    };
+    window.addEventListener('resize', this.resizeHandle);
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.distribution !== this.props.distribution) {
+      this.draw();
+    }
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('resize', this.resizeHandle);
+  }
+
+  draw() {
+    const $scope = this;
+    const { distribution: wrapper, rating } = this.props;
+    if (!wrapper) {
+      return;
+    }
+    const { distribution } = wrapper;
+
+    const colors = [
+      // grey
+      {
+        color: '#9D9FA0',
+        darkerColor: '#9D9FA0',
+        start: 0,
+        end: 899,
+      },
+      // green
+      {
+        color: '#69C329',
+        darkerColor: '#69C329',
+        start: 900,
+        end: 1199,
+      },
+      // blue
+      {
+        color: '#616BD5',
+        darkerColor: '#616BD5',
+        start: 1200,
+        end: 1499,
+      },
+      // yellow
+      {
+        color: '#FCD617',
+        darkerColor: '#FCD617',
+        start: 1500,
+        end: 2199,
+      },
+      // red
+      {
+        color: '#EF3A3A',
+        darkerColor: '#EF3A3A',
+        start: 2200,
+        end: Infinity,
+      },
+    ];
+
+    const desktopMeasurements = {
+      w: 835,
+      h: 400,
+      padding: {
+        top: 20,
+        right: 5,
+        bottom: 100,
+        left: 60,
+      },
+    };
+
+    const mobileMeasurements = {
+      w: 350,
+      h: 200,
+      padding: {
+        top: 50,
+        left: 60,
+        bottom: 80,
+        right: 50,
+      },
+    };
+
+    d3.select($scope.graphRef.current).select('svg').remove();
```

Comment:
Do we really need to use `d3` for rendering? I am fine to rely on `d3` for some auxiliary calculations for axes etc., but the final rendering can be done the same with ReactJS tooling.